### PR TITLE
Add support for google::protobuf::ListValue formatting

### PR DESCRIPTION
**Pull request by itamarkam**
Commit Message: Add support for (Struct) ListValue formatting
Additional Description: Previously, only string and "Struct" were supported. This change mostly refactors some things in the StructFormatter and adds 2 methods for dealing with ListValues.
Risk Level: Low (new functionality that previously threw an exception).
Testing: Unit tests
Docs Changes: N/A
Release Notes: N/A
Platform Specific Features: N/A
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Deprecated:]

### CODE REVIEW

File: test/common/formatter/substitution_formatter_test.cc [hunk 1]
```diff
@@ -1613,6 +1613,38 @@ TEST(SubstitutionFormatterTest, StructFormatterNestedObject) {
   EXPECT_TRUE(TestUtility::protoEqual(out_struct, expected));
 }
 
+TEST(SubstitutionFormatterTest, StructFormatterList) {
```
Comment (yanavlasov) [inline test/common/formatter/substitution_formatter_test.cc:1616]: Can you also add test for list (or struct) nested in a list, please? Similar to the struct formatter tests.

**Review by yanavlasov — CHANGES_REQUESTED** [review]

File: test/common/formatter/substitution_formatter_test.cc (hunk 1 above)
Comment (itamarkam) [reply test/common/formatter/substitution_formatter_test.cc]: Done.

**Comment by itamarkam** [comment]
See latest commit to fix GCC compiling error.

**Review by yanavlasov — APPROVED** [review]

**Pull request merged**